### DISCUSSION TITLE
ESlint Duplicate Identifier Parsing Error

### DISCUSSION THREAD

**Discussion by sumleo** [eslint/eslint#13849] [Q&A]
**Tell us about your environment**

Node version: v14.12.0
npm version: v6.14.8
Local ESLint version: Not found
Global ESLint version: v7.13.0 (Currently used)

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
default
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
var t = 1,
  app = 2;
function t(t) {
  var n;
  for (n in t) return !1;
  return !0;
}
```

```bash
eslint test.js
```

**What did you expect to happen?**
No errors while parsing it (Try it with chrome console)
![image](https://user-images.githubusercontent.com/29517764/99248736-e41da500-2843-11eb-8a2c-5b3268f2caf7.png)

**What actually happened? Please include the actual, raw output from ESLint.**
/Users/test/index/test.js
  3:10  error  Parsing error: Identifier 't' has already been declared

✖ 1 problem (1 error, 0 warnings)

**Are you willing to submit a pull request to fix this bug?**

**Comment by mdjermanovic** [comment]
Hi @sumleo, thanks for the issue!

Can you please share your configuration?

This is a correct parsing error in modules, as top-level functions in modules cannot be redeclared. It works in chrome console since it's parsed as a script.

[Online demo with sourceType: "module"](https://eslint.org/demo#eyJ0ZXh0IjoidmFyIHQgPSAxLFxuICBhcHAgPSAyO1xuZnVuY3Rpb24gdCh0KSB7XG4gIHZhciBuO1xuICBmb3IgKG4gaW4gdCkgcmV0dXJuICExO1xuICByZXR1cm4gITA7XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19) (parsing error)

[Online demo with sourceType: "script"](https://eslint.org/demo#eyJ0ZXh0IjoidmFyIHQgPSAxLFxuICBhcHAgPSAyO1xuZnVuY3Rpb24gdCh0KSB7XG4gIHZhciBuO1xuICBmb3IgKG4gaW4gdCkgcmV0dXJuICExO1xuICByZXR1cm4gITA7XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19) (no parsing error)

**Comment by eslint-deprecated** [comment]
Hi @sumleo, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

[//]: # (needs-info)
